src/k8s_client/k8s_client.go

```diff
@@ -627,3 +629,42 @@ func (c *k8sClient) IsClusterCapabilityEnabled(capability configv1.ClusterVersio
 	isKnown := funk.Contains(version.Status.Capabilities.KnownCapabilities, capability)
 	return isExplicitlyEnabled || !isKnown, nil
 }
+
+type taintRemovePatch struct {
+	Op   string `json:"op"`
+	Path string `json:"path"`
+}
+
+func (c *k8sClient) UntaintNode(name string) error {
+	node, err := c.client.CoreV1().Nodes().Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	foundTaintIndex := -1
+	for i, taint := range node.Spec.Taints {
+		if taint.Key == UNINITIALIZED_TAINT_KEY {
+			foundTaintIndex = i
+		}
+	}
```

Comment:
```suggestion
	uninitialized := slices.Contains(node.Spec.Taints, UNINITIALIZED_TAINT_KEY)
```
https://pkg.go.dev/k8s.io/utils/strings/slices

Reply:
1. This needs to be an exact Taint match (including value and operator)
2. Later on we need the found taint index to remove it via JSON patch